### PR TITLE
Fix for latest elixir

### PR DESCRIPTION
**Pull request by felipesere**
Its a quick update to make work under a newer version of Elixir.

The Access protocol was deprecated so I just moved the functions into the Module.

### CODE REVIEW

**Comment by jfacorro** [comment]
Hi @felipesere. Than you for your contribution! Please notice that Travis CI is complaining about the Elixir version which should be updated in the [`.travis.yml`](https://github.com/jfacorro/Eden/blob/master/.travis.yml) file.

File: test/eden_test.exs [hunk 1]
```diff
@@ -122,16 +122,6 @@ defmodule EdenTest do
     assert encode!(set) == "#\{:name, :age, \"John\", 42}"
   end
 
-  test "Encode Tag" do
```
Comment (jfacorro) [inline test/eden_test.exs:125]: Why did you delete this test?

Comment (felipesere) [inline test/eden_test.exs:125]: Because I got this failure:

```
test Encode Tag (EdenTest)
     test/eden_test.exs:125
     Assertion with == failed
     code: encode!(date) == "#inst \"1985-04-12T23:20:50.052Z\""
     lhs:  "#inst \"1985-04-12T23:20:50.520Z\""
     rhs:  "#inst \"1985-04-12T23:20:50.052Z\""
     stacktrace:
       test/eden_test.exs:127
```

which I now see is just a typo. I brushed it off as a timing. issue.

**Comment by jfacorro** [comment]
@felipesere Thank you fixing the bug in the tests. Travis CI build is still failing though. Could you update Elixir's version in the `.travis.yml` file? Once you do that I can merge the PR.

**Comment by felipesere** [comment]
For some reason it fails with OTP 17.1 due to some issue with `rebar`:

```
Uncaught error in rebar_core: {'EXIT',
                               {{badmatch,{error,eexist}},
                                [{rebar_deps,update_deps_code_path,2,[]},
                                 {rebar_deps,preprocess,2,[]},
                                 {rebar_core,acc_modules,5,[]},
                                 {rebar_core,process_dir1,7,[]},
                                 {rebar_core,process_commands,2,[]},
                                 {rebar,main,1,[]},
                                 {escript,run,2,
                                  [{file,"escript.erl"},{line,752}]},
                                 {escript,start,1,
                                  [{file,"escript.erl"},{line,276}]}]}}
==> eden
```

Does it tell you anything?

**Comment by jfacorro** [comment]
@felipesere The problem is there is a [bug](http://erlang.org/pipermail/erlang-bugs/2014-July/004484.html) in the `filelib` module for `17.1` which causes an error in this [function call](https://github.com/rebar/rebar/blob/365ac649dc818619757f96a699ddb174f004cff9/src/rebar_deps.erl#L337). `rebar3` even [shows a message](https://github.com/rebar/rebar3/blob/fe470e534ccc22d153ca9005d7dd4c1f85447140/src/rebar3.erl#L89) when compiling with `17.1`.

**Comment by jfacorro** [comment]
@felipesere Let's remove 17.1 since I don't see a way we can solve this issue without some of the deps being modified.
